.github/workflows/ci.yml

```
@@ -13,159 +10,78 @@ on:
-        run: pnpm store status
-
-  lint:
-    needs: pnpmstore
-    timeout-minutes: 5
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ ubuntu-latest ]
-        node: [ 14 ]
-    steps:
+      - run: npm i -g pnpm@6
       - uses: actions/setup-node@v2
         with:
           node-version: ${{ matrix.node }}
-
-      - name: checkout
-        uses: actions/checkout@v2
-
-      - name: pnpm-store
-        uses: actions/cache@v2
-        id: pnpm-store
-        with:
-          path: ${{ env.pnpm_store_path }}
-          key: ${{ matrix.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
-      - name: install pnpm
-        run: npm i -g pnpm@6
-      - name: set pnpm store-dir
-        run: pnpm config set store-dir ${{ env.pnpm_store_path }}
+          cache: 'pnpm'
+          cache-dependency-path: '**/pnpm-lock.yaml'
       - name: install
-        run: pnpm install --frozen-lockfile --offline --ignore-scripts
-      - name: install esbuild
-        run: node node_modules/esbuild/install.js
+        run: |
+          pnpm install --frozen-lockfile --prefer-offline --ignore-scripts
+          node node_modules/esbuild/install.js
       - name: build
-        run: pnpm run build:ci
+        id: build
+        run: pnpm run build
       - name: lint
-        run: pnpm lint
-
-  audit:
-    needs: pnpmstore
-    timeout-minutes: 5
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ ubuntu-latest ]
-        node: [ 14 ]
-    steps:
-      - uses: actions/setup-node@v2
-        with:
-          node-version: ${{ matrix.node }}
-
-      - name: checkout
-        uses: actions/checkout@v2
-
-      - name: pnpm-store
-        uses: actions/cache@v2
-        id: pnpm-store
-        with:
-          path: ${{ env.pnpm_store_path }}
-          key: ${{ matrix.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
-      - name: install pnpm
-        run: npm i -g pnpm@6
-      - name: set pnpm store-dir
-        run: pnpm config set store-dir ${{ env.pnpm_store_path }}
-      - name: install
-        run: pnpm install --frozen-lockfile --offline --ignore-scripts
+        if: (${{ success() }} || ${{ failure() }})
+        run: pnpm run lint
       - name: audit
+        if: (${{ success() }} || ${{ failure() }})
         run: pnpm audit
 
+  # this is the test matrix, it runs with node14 on linux,windows,macos + node12,16 on linux
+  # it is skipped if the build step of the checks job wasn't successful (still runs if lint or audit fail)
   test:
-    needs: pnpmstore
+    needs: checks
+    if: (${{ success() }} || ${{ failure() }}) && (${{ needs.checks.output.build_successful }})
     timeout-minutes: 10
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os: [ ubuntu-latest, windows-latest, macos-latest ]
-        node: [ 12, 14, 16 ]
+        node-version: [14]
+        os: [ubuntu-latest, macOS-latest, windows-latest]
```

```suggestion
        os: [ubuntu-latest, macos-latest, windows-latest]
```

Would be nice to have macos small caps so github makes a cleaner output :grin:

---

done